lib/system/dollars.nim

```diff
@@ -75,33 +75,28 @@ proc `$`*[T: tuple|object](x: T): string =
   ##   $(a: 23, b: 45) == "(a: 23, b: 45)"
   ##   $() == "()"
   result = "("
-  var firstElement = true
-  const isNamed = T is object or isNamedTuple(T)
-  when not isNamed:
-    var count = 0
+  const isNamed = x is object or isNamedTuple(T)
+  var count = 0
   for name, value in fieldPairs(x):
-    if not firstElement: result.add(", ")
-    when isNamed:
+    if count != 0:
+      result.add(", ")
+    if isNamed:
       result.add(name)
       result.add(": ")
-    else:
-      count.inc
+
     when compiles($value):
-      when value isnot string and value isnot seq and compiles(value.isNil):
-        if value.isNil: result.add "nil"
```

Comment:
For custom types that have their own implementation of ``isNil`` this changes the existing behavior. Apart from that I like the simplification via the ``count`` variable.

Reply:
Should I revert to ``isNil`` then? I thought this ``isNil`` is this ancient thing that was once necessary, but by now we can live a happly live without ever using it?

Reply:
Yes, keep the ``isNil`` and the ``compiles(value.isNil)``.